### ISSUE TITLE
scheduler and controller-manager spamming logs with: "lock is held by <server> and has not yet expired"

### ISSUE THREAD

**Issue by jonaz** [kubernetes/kubernetes#34594]
**What keywords did you search in Kubernetes issues before filing this one?** (If you have found any duplicates, you should instead reply there.): "lock is held by"  "has not yet expired"

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): BUG REPORT

**Kubernetes version** (use `kubectl version`):
tested on 1.3.5

```
Client Version: version.Info{Major:"1", Minor:"3", GitVersion:"v1.3.0", GitCommit:"283137936a498aed572ee22af6774b6fb6e9fd94", GitTreeState:"clean", BuildDate:"2016-07-01T19:26:38Z", GoVersion:"go1.6.2", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"3", GitVersion:"v1.3.5+coreos.1", GitCommit:"4fc64a21f8a593d97b75296acd1e979391518b28", GitTreeState:"clean", BuildDate:"2016-08-22T17:15:28Z", GoVersion:"go1.6.2", Compiler:"gc", Platform:"linux/amd64"}
```

and 1.3.8 (upgraded today)

```
Client Version: version.Info{Major:"1", Minor:"3", GitVersion:"v1.3.0", GitCommit:"283137936a498aed572ee22af6774b6fb6e9fd94", GitTreeState:"clean", BuildDate:"2016-07-01T19:26:38Z", GoVersion:"go1.6.2", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"3", GitVersion:"v1.3.8+coreos.1", GitCommit:"6b29caf6507ce9989984720212fd3e3341069e06", GitTreeState:"clean", BuildDate:"2016-09-30T16:54:50Z", GoVersion:"go1.6.2", Compiler:"gc", Platform:"linux/amd64"}
```

**Environment**:
-  Hardware: vmware guest
-  OS: CoreOs 1122.2.0
-  kernel: 4.7.0-coreos
- masters: 3

/etc/kubernetes/manifests/kube-scheduler.yaml looks like this:

```
apiVersion: v1
kind: Pod
metadata:
  name: kube-scheduler
  namespace: kube-system
  labels:
    app: kube-scheduler
spec:
  hostNetwork: true
  containers:
  - name: kube-scheduler
    image: quay.io/coreos/hyperkube:v1.3.5_coreos.1
    command:
    - /hyperkube
    - scheduler
    - --master=http://127.0.0.1:8080
    - --leader-elect=true
    - --v=2
    livenessProbe:
      httpGet:
        host: 127.0.0.1
        path: /healthz
        port: 10251
      initialDelaySeconds: 15
      timeoutSeconds: 1
```

/etc/kubernetes/manifests/kube-controller-manager.yaml looks like this:

```
apiVersion: v1
kind: Pod
metadata:
  name: kube-controller-manager
  namespace: kube-system
  labels:
    app: kube-controller-manager
spec:
  hostNetwork: true
  containers:
  - name: kube-controller-manager
    image: quay.io/coreos/hyperkube:v1.3.5_coreos.1
    command:
    - /hyperkube
    - controller-manager
    - --master=http://127.0.0.1:8080
    - --leader-elect=true 
    - --service-account-private-key-file=/etc/kubernetes/ssl/apiserver-key.pem
    - --root-ca-file=/etc/kubernetes/ssl/ca.pem
    - --v=2
    livenessProbe:
      httpGet:
        host: 127.0.0.1
        path: /healthz
        port: 10252
      initialDelaySeconds: 15
      timeoutSeconds: 1
    volumeMounts:
    - mountPath: /etc/kubernetes/ssl
      name: ssl-certs-kubernetes
      readOnly: true
    - mountPath: /etc/ssl/certs
      name: ssl-certs-host
      readOnly: true
  volumes:
  - hostPath:
      path: /etc/kubernetes/ssl
    name: ssl-certs-kubernetes
  - hostPath:
      path: /usr/share/ca-certificates
    name: ssl-certs-host

```

**What happened**:

The logs from kube-scheduler and kube-controller-manager are spamming with:

```
I1012 07:13:30.700862       1 leaderelection.go:296] lock is held by utv-kube03 and has not yet expired
I1012 07:13:33.263266       1 leaderelection.go:296] lock is held by utv-kube03 and has not yet expired
I1012 07:13:36.944429       1 leaderelection.go:296] lock is held by utv-kube03 and has not yet expired
I1012 07:13:40.486944       1 leaderelection.go:296] lock is held by utv-kube03 and has not yet expired
I1012 07:13:44.772153       1 leaderelection.go:296] lock is held by utv-kube03 and has not yet expired
I1012 07:13:47.186971       1 leaderelection.go:296] lock is held by utv-kube03 and has not yet expired
I1012 07:13:49.914362       1 leaderelection.go:296] lock is held by utv-kube03 and has not yet expired
I1012 07:13:52.353564       1 leaderelection.go:296] lock is held by utv-kube03 and has not yet expired
I1012 07:13:56.181010       1 leaderelection.go:296] lock is held by utv-kube03 and has not yet expired
I1012 07:13:59.263994       1 leaderelection.go:296] lock is held by utv-kube03 and has not yet expired
I1012 07:14:03.553559       1 leaderelection.go:296] lock is held by utv-kube03 and has not yet expired
I1012 07:14:06.325838       1 leaderelection.go:296] lock is held by utv-kube03 and has not yet expired
I1012 07:14:10.595729       1 leaderelection.go:296] lock is held by utv-kube03 and has not yet expired
I1012 07:14:13.347343       1 leaderelection.go:296] lock is held by utv-kube03 and has not yet expired
I1012 07:14:17.330717       1 leaderelection.go:296] lock is held by utv-kube03 and has not yet expired
I1012 07:14:21.505816       1 leaderelection.go:296] lock is held by utv-kube03 and has not yet expired
I1012 07:14:23.950507       1 leaderelection.go:296] lock is held by utv-kube03 and has not yet expired
I1012 07:14:26.892771       1 leaderelection.go:296] lock is held by utv-kube03 and has not yet expired
I1012 07:14:30.914393       1 leaderelection.go:296] lock is held by utv-kube03 and has not yet expired
I1012 07:14:34.408769       1 leaderelection.go:296] lock is held by utv-kube03 and has not yet expired
I1012 07:14:38.544271       1 leaderelection.go:296] lock is held by utv-kube03 and has not yet expired
I1012 07:14:40.645166       1 leaderelection.go:296] lock is held by utv-kube03 and has not yet expired
I1012 07:14:43.919125       1 leaderelection.go:296] lock is held by utv-kube03 and has not yet expired
I1012 07:14:47.663210       1 leaderelection.go:296] lock is held by utv-kube03 and has not yet expired
I1012 07:14:51.768576       1 leaderelection.go:296] lock is held by utv-kube03 and has not yet expired
I1012 07:14:55.748494       1 leaderelection.go:296] lock is held by utv-kube03 and has not yet expired
I1012 07:15:00.116892       1 leaderelection.go:296] lock is held by utv-kube03 and has not yet expired
I1012 07:15:03.489400       1 leaderelection.go:296] lock is held by utv-kube03 and has not yet expired
I1012 07:15:06.465890       1 leaderelection.go:296] lock is held by utv-kube03 and has not yet expired
I1012 07:15:09.703411       1 leaderelection.go:296] lock is held by utv-kube03 and has not yet expired
I1012 07:15:13.463524       1 leaderelection.go:296] lock is held by utv-kube03 and has not yet expired
I1012 07:15:15.607621       1 leaderelection.go:296] lock is held by utv-kube03 and has not yet expired
I1012 07:15:17.896426       1 leaderelection.go:296] lock is held by utv-kube03 and has not yet expired
I1012 07:15:22.087427       1 leaderelection.go:296] lock is held by utv-kube03 and has not yet expired
I1012 07:15:26.217274       1 leaderelection.go:296] lock is held by utv-kube03 and has not yet expired
I1012 07:15:30.269665       1 leaderelection.go:296] lock is held by utv-kube03 and has not yet expired
I1012 07:15:33.637867       1 leaderelection.go:296] lock is held by utv-kube03 and has not yet expired
I1012 07:15:36.712575       1 leaderelection.go:296] lock is held by utv-kube03 and has not yet expired
I1012 07:15:40.783867       1 leaderelection.go:296] lock is held by utv-kube03 and has not yet expired
I1012 07:15:44.372188       1 leaderelection.go:296] lock is held by utv-kube03 and has not yet expired
I1012 07:15:47.605168       1 leaderelection.go:296] lock is held by utv-kube03 and has not yet expired
I1012 07:15:50.434502       1 leaderelection.go:296] lock is held by utv-kube03 and has not yet expired
I1012 07:15:53.216699       1 leaderelection.go:296] lock is held by utv-kube03 and has not yet expired
I1012 07:15:57.015595       1 leaderelection.go:296] lock is held by utv-kube03 and has not yet expired
I1012 07:15:59.152636       1 leaderelection.go:296] lock is held by utv-kube03 and has not yet expired
I1012 07:16:01.301530       1 leaderelection.go:296] lock is held by utv-kube03 and has not yet expired
```

The node holding the lock only logs this:

```
I1012 07:27:20.982437       1 leaderelection.go:215] sucessfully acquired lease kube-system/kube-scheduler
```

**What you expected to happen**:
If this is expected behaviour i would like to be able to supress those messages with --v=2

**How to reproduce it** (as minimally and precisely as possible):
I think it would be sufficient to run scheduler or controller-manager with `--leader-elect?true`. 

etcdctl get registry/services/endpoints/kube-system/kube-controller-manager looks like this:

```
{
    "kind": "Endpoints",
    "apiVersion": "v1",
    "metadata": {
        "name": "kube-controller-manager",
        "namespace": "kube-system",
        "selfLink": "/api/v1/namespaces/kube-system/endpoints/kube-controller-manager",
        "uid": "b9ee7a6f-8f82-11e6-b186-0050568802e1",
        "creationTimestamp": "2016-10-11T07:17:10Z",
        "annotations": {
            "control-plane.alpha.kubernetes.io/leader": "{\"holderIdentity\":\"utv-kube03\",\"leaseDurationSeconds\":15,\"acquireTime\":\"2016-10-11T07:17:10Z\",\"renewTime\":\"2016-10-12T07:18:54Z\",\"leaderTransitions\":0}"
        }
    },
    "subsets": []
}
```

The cluster seems healthy and are scheduling pods correctly.

**Comment by jonaz** [comment]
Same issue with 1.4.4

**Comment by jaygorrell** [comment]
Just hit this with 1.4.6 as well.

**Comment by hsyed** [comment]
The spam is overwhelming please move it to an appropriate log level.

**Comment by timothysc** [comment]
This is expected behavior, this appears from standby controllers and schedulers.  You can tune these parameter on the binaries.. see leader-election params under --help.  

closing, not an bug

**Comment by jonaz** [comment]
@timothysc i cannot find any way to supress this exept turning upp the duration and thats not what we want. I only dont want to fill elasticerarch with those messages.

https://github.com/kubernetes/kubernetes/blob/master/pkg/client/leaderelection/leaderelection.go#L247

Why dont they have a proper loglevel?

**Comment by timothysc** [comment]
K, I'll adjust the log levels.

**Comment by weitzj** [comment]
@timothysc Does the controller-manager start with `-v=0` by default?

Did I understand this correctly that you moved it to `-v=4` ?

**Comment by timothysc** [comment]
> Does the controller-manager start with -v=0 by default? 

default depends on provider, some folks default to 3.  Needless to say -v 4 "should be" outside the normal range to prevent spamming. 

> Does this by chance have anything to do with the log for Ingress controllers as well and will be fixed in 1.6 ? 

verbosity on ingress controller is controlled by how that component is started, but I applied it to the base so it should hit *.
